File: lib/ts/src/internal/types/THeaders.ts
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

import { TFieldDeclaration } from './TFieldDeclaration.js';

export class THeaders {
    name: string;
    requestHeaders: { [key: string]: TFieldDeclaration };
    responseHeaders: { [key: string]: TFieldDeclaration };

    constructor(
        name: string,
        requestHeaders: { [key: string]: TFieldDeclaration },
        responseHeaders: { [key: string]: TFieldDeclaration },
    ) {
        this.name = name;
        this.requestHeaders = requestHeaders;
        this.responseHeaders = responseHeaders;
    }
}

File: lib/go/internal/types/THeaders.go
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

package types

// THeaders captures request/response header declarations for a function.
type THeaders struct {
	Name            string
	RequestHeaders  map[string]*TFieldDeclaration
	ResponseHeaders map[string]*TFieldDeclaration
}

// NewTHeaders constructs a THeaders instance.
func NewTHeaders(name string, requestHeaders, responseHeaders map[string]*TFieldDeclaration) *THeaders {
	return &THeaders{
		Name:            name,
		RequestHeaders:  requestHeaders,
		ResponseHeaders: responseHeaders,
	}
}
